provider/maas/environ.go

```diff
@@ -859,6 +861,24 @@ func (*maasEnviron) MaintainInstance(args environs.StartInstanceParams) error {
 	return nil
 }
 
+type statusCallbackFunc func(settableStatus status.Status, info string, data map[string]interface{}) error
+
+func pollinstanceStatus(done <-chan struct{}, statusCallback statusCallbackFunc, inst maasInstance) {
+	for {
+		instStatus := inst.Status()
+		err := statusCallback(instStatus.Status, instStatus.Message, nil)
+		if err != nil {
+			logger.Errorf("setting instance %s status: %v", inst.Id(), err)
+		}
+
+		select {
+		case <-done:
+			return
+		case <-time.After(statusPollInterval):
```

Comment:
If that's the case, I disagree. It doesn't have to be all or nothing; we can request it from the caller, and for now -- since we don't have time -- just pass in WallClock. It edges us forward.